integration/e2e/chaincode_service_test.go

```diff
@@ -0,0 +1,275 @@
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Chaincode As External Service Suite")
+}
+
+var _ = Describe("ChaincodeAsExternalService", func() {
+	var (
+		testDir                 string
+		network                 *nwo.Network
+		extcc                   nwo.Chaincode
+		chaincodeServerAddrress string
+		certFiles               []string
+		process                 ifrit.Process
+		extbldr                 fabricconfig.ExternalBuilder
+	)
+
+	BeforeEach(func() {
+		var err error
+		testDir, err = ioutil.TempDir("", "e2e-chaincode-service")
+		Expect(err).NotTo(HaveOccurred())
+		extcc = nwo.Chaincode{
+			Name:            "mycc",
+			Version:         "0.0",
+			Path:            components.Build("github.com/hyperledger/fabric/integration/chaincode/extcc"),
+			Lang:            "extcc",
+			PackageFile:     filepath.Join(testDir, "extcc.tar.gz"),
+			Ctor:            `{"Args":["init","a","100","b","200"]}`,
+			SignaturePolicy: `AND ('Org1MSP.member','Org2MSP.member')`,
+			Sequence:        "1",
+			InitRequired:    true,
+			Label:           "my_extcc_chaincode",
+		}
+
+		cwd, err := os.Getwd()
+		Expect(err).NotTo(HaveOccurred())
+		extbldr = fabricconfig.ExternalBuilder{
+			Path: filepath.Join(cwd, "..", "externalbuilders", "extcc"),
+			Name: "extcc",
+		}
+
+	})
+
+	AfterEach(func() {
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+		if network != nil {
+			network.Cleanup()
+		}
+		os.RemoveAll(testDir)
+	})
+
+	Describe("basic solo network with 2 orgs and external chaincode service", func() {
+		var (
+			datagramReader *DatagramReader
+			ccserver       ifrit.Process
+		)
+
+		BeforeEach(func() {
+			datagramReader = NewDatagramReader()
+			go datagramReader.Start()
+
+			//network = nwo.New(nwo.BasicSoloWithChaincodeServers(&extcc), testDir, nil, StartPort(), components)
+			network = nwo.New(nwo.BasicSolo(), testDir, nil, StartPort(), components)
+
+			chaincodeServerAddrress = fmt.Sprintf("127.0.0.1:%d", network.ReservePort())
+
+			tlsCA, err := tlsgen.NewCA()
+			Expect(err).NotTo(HaveOccurred())
+			certFiles = generateCertKeysAndConnectionFiles(tlsCA, chaincodeServerAddrress, testDir, extcc.Name)
+			generateClientConnectionFile(tlsCA, chaincodeServerAddrress, testDir, extcc.Name)
+
+			//add extcc builder
+			network.ExternalBuilders = append(network.ExternalBuilders, extbldr)
+
+			network.MetricsProvider = "statsd"
+			network.StatsdEndpoint = datagramReader.Address()
+			network.Profiles = append(network.Profiles, &nwo.Profile{
+				Name:          "TwoOrgsBaseProfileChannel",
+				Consortium:    "SampleConsortium",
+				Orderers:      []string{"orderer"},
+				Organizations: []string{"Org1", "Org2"},
+			})
+			network.Channels = append(network.Channels, &nwo.Channel{
+				Name:        "baseprofilechannel",
+				Profile:     "TwoOrgsBaseProfileChannel",
+				BaseProfile: "TwoOrgsOrdererGenesis",
+			})
```

Comment:
I don't understand why we need to augment the network before generating the config. I don't see anything in here referring to `baseprofilechannel`. Can you please elaborate on why this is needed or modify the test to use the out of the box `BasicSolo` config?